### PR TITLE
Fix QueueEntry recycle problem.

### PR DESCRIPTION
**Pull request by horizonzy**
Descriptions of the changes in this PR:
In the QueueEntry recycle, it only recycles itself to the object pool, but didn't reset some properties.
Like entry, cb, etc. We should reset the filed before recycles itself.

### CODE REVIEW

**Review by hangc0276 — APPROVED** [review]
Nice catch!

**Review by nicoloboschi — APPROVED** [review]

**Review by StevenLuMT — APPROVED** [review]
Great work

**Review by wenbingshen — APPROVED** [review]
nice work!

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java [hunk 1]
```diff
@@ -347,8 +347,8 @@ public void run() {
             cbThreadPoolQueueSize.dec();
             journalAddEntryStats.registerSuccessfulEvent(MathUtils.elapsedNanos(enqueueTime), TimeUnit.NANOSECONDS);
             cb.writeComplete(0, ledgerId, entryId, null, ctx);
-            recycle();
             callbackTime.addLatency(MathUtils.elapsedNanos(startTime), TimeUnit.NANOSECONDS);
+            recycle();
```
Comment (lhotari) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java:351]: @horizonzy This change wasn't mentioned in the PR description. Why was this missing before? Was the recycling completely unused before? /cc @hangc0276

Comment (horizonzy) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java:351]: Cause we already reset `callbackTime` to null. And we should ensure do the all works before recycle self.

**Pull request merged**